schematic/store/synapse.py

```diff
@@ -1382,34 +1493,27 @@ def buildDB(
             manifest_table_id: synID of the uploaded table
 
         """
-        table_info = self.get_table_info(datasetId=datasetId)
-        # Put table manifest onto synapse
-        schema = Schema(
-            name=table_name,
-            columns=col_schema,
-            parent=self.getDatasetProject(datasetId),
+        table_parent_id = self.getDatasetProject(datasetId=datasetId)
+        existing_table_id = self.syn.findEntityId(
+            name=table_name, parent=table_parent_id
         )
```

Comment:
Swapping to `findEntityId` prevents the need to walk the project to find the applicable table ID. Instead we can utilize the Synapse Rest API to find the ID for us.

Reply:
This will return `None` if not found